inst/include/dust/rng.hpp

```diff
@@ -12,67 +12,82 @@ namespace dust {
 
 // This is just a container class for state
 template <typename T>
-class pRNG { // # nocov
+class pRNG {
 public:
-  pRNG(const size_t n, const std::vector<uint64_t>& seed) {
-    rng_state_t<T> s;
+  pRNG(const size_t n, const std::vector<uint64_t>& seed) :
+    n_(n), state_(n * rng_state_t<T>::size()) {
     auto len = rng_state_t<T>::size();
     auto n_seed = seed.size() / len;
     for (size_t i = 0; i < n; ++i) {
+      rng_state_t<T> s = state(i);
       if (i < n_seed) {
-        std::copy_n(seed.begin() + i * len, len, s.state.begin());
+        for (size_t j = 0; j < len; ++j) {
+          s[j] = seed[i * len + j];
+        }
       } else {
+        rng_state_t<T> prev = state(i - 1);
+        for (size_t j = 0; j < len; ++j) {
+          s[j] = prev[j];
+        }
```

Comment:
This isn't interleaved, is that right? Or is it coming from xoshiro where it is already interleaved?

Reply:
Added a comment now to clarify. You are right, it's not interleaved and this is only ever initialised from R